FirebaseInAppMessaging/Sources/FIRInAppMessaging.m

```diff
@@ -53,38 +53,36 @@ + (void)load {
       ^id _Nullable(FIRComponentContainer *container, BOOL *isCacheable) {
     // Ensure it's cached so it returns the same instance every time fiam is called.
     *isCacheable = YES;
+
+    // Only configure for the default FIRApp.
+    if (!container.app.isDefaultApp) {
+      FIRLogError(kFIRLoggerInAppMessaging, @"I-IAM170000",
+                  @"In-App Messaging must be used with the default Firebase app.");
+      return nil;
+    }
+
     id<FIRAnalyticsInterop> analytics = FIR_COMPONENT(FIRAnalyticsInterop, container);
     FIRInstallations *installations = [FIRInstallations installationsWithApp:container.app];
+
+    if (_autoBootstrapOnFIRAppInit) {
+      FIRLogDebug(kFIRLoggerInAppMessaging, @"I-IAM170002",
+                  @"Auto bootstrap Firebase in-app messaging SDK");
+      [FIRInAppMessaging bootstrapIAMFromFIRApp:container.app];
+    } else {
+      FIRLogDebug(kFIRLoggerInAppMessaging, @"I-IAM170003",
+                  @"No auto bootstrap Firebase in-app messaging SDK");
+    }
+
     return [[FIRInAppMessaging alloc] initWithAnalytics:analytics installations:installations];
   };
   FIRComponent *fiamProvider =
       [FIRComponent componentWithProtocol:@protocol(FIRInAppMessagingInstanceProvider)
-                      instantiationTiming:FIRInstantiationTimingLazy
+                      instantiationTiming:FIRInstantiationTimingEagerInDefaultApp
```

Comment:
I wonder if we want something like LeafEager, to insure that it runs after things like Auth and Analytics that are time critical to run early?